src/main/java/com/michelin/cio/hudson/plugins/maskpasswords/MaskPasswordsOutputStream.java

```diff
@@ -49,22 +49,36 @@
      *               will write to
      * @param passwords A collection of {@link String}s to be masked
      */
-    public MaskPasswordsOutputStream(OutputStream logger, Collection<String> passwords) {
+    public MaskPasswordsOutputStream(OutputStream logger, Collection<String> passwords, Collection<String> regexes) {
         this.logger = logger;
 
-        if(passwords != null && passwords.size() > 0) {
+
+        if((passwords != null && passwords.size() > 0) || (regexes != null && regexes.size() > 0)) {
             // passwords are aggregated into a regex which is compiled as a pattern
             // for efficiency
             StringBuilder regex = new StringBuilder().append('(');
 
             int nbMaskedPasswords = 0;
-            for(String password: passwords) {
-                if(StringUtils.isNotEmpty(password)) { // we must not handle empty passwords
-                    regex.append(Pattern.quote(password));
-                    regex.append('|');
-                    nbMaskedPasswords++;
-                }
+
+            if(passwords != null && passwords.size() > 0) {
+              for(String password: passwords) {
+                  if(StringUtils.isNotEmpty(password)) { // we must not handle empty passwords
+                      regex.append(Pattern.quote(password));
+                      regex.append('|');
+                      nbMaskedPasswords++;
+                  }
+              }
             }
+            if(regexes != null && regexes.size() > 0) {
+              for(String user_regex: regexes) {
+                  if(StringUtils.isNotEmpty(user_regex)) { // we must not handle empty passwords
+                      regex.append(user_regex);
```

Comment:
I don't see how the regex could be escaped and still function as a regex - the whole point of this for me is to be able to mask an actual regex - i.e. if the user wants to mask ``((f|o)o [Bb]ar)`` they should be able to.

I've explicitly documented how this works in the [help for the global var](https://github.com/manheim/mask-passwords-plugin/blob/mask-regex-global-clf/src/main/resources/com/michelin/cio/hudson/plugins/maskpasswords/MaskPasswordsBuildWrapper/help-globalVarMaskRegexes.html#L25) but I can also add similar help to the per-job config.

For my use case at least, escaping isn't going to be possible; the regex that we want to mask is ``(?<![A-Za-z0-9/+=])[A-Za-z0-9/+=]{40}(?![A-Za-z0-9/+=])``

Maybe I should just change this to something like ``regex.append("(" + user_regex + ")");`` ?